internal/app/backend/backend_service.go

```diff
@@ -330,6 +339,64 @@ func (s *backendService) AssignTickets(ctx context.Context, req *pb.AssignTicket
 	return resp, nil
 }
 
+func handleMatch(ctx context.Context, match *pb.Match, store statestore.Service) error {
+	backfill := match.GetBackfill()
+
+	if backfill == nil {
+		return nil
+	}
+
+	ticketdIds := make([]string, len(match.Tickets))
+
+	for _, t := range match.Tickets {
+		ticketdIds = append(ticketdIds, t.Id)
+	}
+
+	if backfill.Id != "" {
```

Comment:
This leans in to style and it's totally optional, but consider:

Often when an if statement has both clauses used, (in this case, the rest of this function could be in an "else" clause with no change to behavior) it's better to have the if statement be the "positive" clause.  That is, `if backfill.Id == ""`.  That way when you get to the "else" clause, you're not dealing with a double negative mentally.

Also, when doing that on this function, I think `doCreateBackfill` and `doUpdateBackfill` should be inlined.  It makes for a long function, but that's ok because it's just doing a bunch of stuff that needs to be done in order.   A comment works just as well as a function name, and inlining removes overall complexity (fewer functions.)

(It helps in this case `doCreateBackfill` is relatively short, doing the swap on `if` and then inlining would happen to be relatively clean.  If the length of the results were flipped, I'd probably prefer inlining over simplifying the if statements.  Here we can get best of both worlds, I think...)

Reply:
Interesting point of view. Inlined functions and used positive checks.